### PR TITLE
Fix masked composites with negative offsets

### LINKED ISSUE #5477: negative x,y in ImageCompositeMasked produces a tensor error
### Expected Behavior

negative x&y values should place the source rectangle off the left and top of the image, the same as the source rectangle can go off the right and bottom of the image already

### Actual Behavior

gets an error with mismatched tensor sizes, because the code uses `[left:right]` to slice destination, but if x is negative, left is negative, so `left` in slice notation is treated as an offset from the end rather than a negative position, so the length of the resulting tensor is nonsense.

### Steps to Reproduce

The code seems to be intended to support negative x & y, given the appearance of `min(0, x)` etc. in various places. That's why this is a bug report not a feature request.

A complete fix is as follows:
```
    # calculate the bounds of the source that will be overlapping the destination
    # this prevents the source trying to overwrite latent pixels that are out of bounds
    # of the destination
    visible_width, visible_height = (destination.shape[3] - left + min(0, x), destination.shape[2] - top + min(0, y),)

>   offset_x, offset_y = max(0,-x//multiplier),max(0,-y//multiplier)
>   mask = mask[:, :, offset_y:visible_height, offset_x:visible_width]
    inverse_mask = torch.ones_like(mask) - mask

>   source_portion = mask * source[:, :, offset_y:visible_height, offset_x:visible_width]

>   left,top = max(left,0),max(top,0)
    destination_portion = inverse_mask  * destination[:, :, top:bottom, left:right]

    destination[:, :, top:bottom, left:right] = source_portion + destination_portion
    return destination
```

In addition, the node description needs to be updated to allow negative x & y, for which there is an existing but closed pull request #2574  (although you can already use negative values if you make them inputs).

### Debug Logs

```powershell
.
```

### Other

_No response_

### PR DESCRIPTION
**Pull request by yzxcj797**
## Summary
- Clip the source, mask, and destination to their actual intersection before blending.
- Preserve correct output for negative, positive, and non-overlapping offsets.
- Allow negative X/Y widget values for both image and latent masked composites.

Fixes #5477

## Tests
- `pytest tests-unit/comfy_extras_test/nodes_mask_test.py -q`
- `pytest tests-unit/comfy_extras_test -q --ignore=tests-unit/comfy_extras_test/compositor_node_test.py --ignore=tests-unit/comfy_extras_test/nodes_math_test.py --ignore-glob=tests-unit/comfy_extras_test/test_seedvr2_*.py`
- `ruff check comfy_extras/nodes_mask.py tests-unit/comfy_extras_test/nodes_mask_test.py`
- `git diff --check`

### CODE REVIEW

File: comfy_extras/nodes_mask.py [hunk 1]
```diff
@@ -29,21 +29,26 @@ def composite(destination, source, x, y, mask = None, multiplier = 8, resize_sou
...
 
     inverse_mask = torch.ones_like(mask) - mask
 
-    source_portion = mask * source[..., :visible_height, :visible_width]
-    destination_portion = inverse_mask  * destination[..., top:bottom, left:right]
+    source_portion = mask * source[..., source_top:source_bottom, source_left:source_right]
+    destination_portion = inverse_mask * destination[..., destination_top:destination_bottom, destination_left:destination_right]
 
-    destination[..., top:bottom, left:right] = source_portion + destination_portion
+    destination[..., destination_top:destination_bottom, destination_left:destination_right] = source_portion + destination_portion
```
[the remark this answers is not in the document]
Comment (yzxcj797) [reply comfy_extras/nodes_mask.py:51]: The existing clamp prevents a negative destination end: for a 2-high source, `y=-4` becomes `-2`, so the destination height range is `0:0` and the source height range is `2:2`—both intersections are empty. More generally, `y` is clamped to at least `-source.height * multiplier`, so `bottom` cannot be negative.

Commit a9d77ef8 changes the no-overlap regression to use an explicit all-ones mask, covering the masked path called out here; it passes locally.
